### PR TITLE
Use SourceParser on primary key lookups

### PR DESCRIPTION
**Pull request by seut**
By using the source parser, only required columns are parsed instead of parsing the complete source into a map.
It will improve performance on PK lookups slightly as seen in the benchmark. Performance is expected to increase on larger documents while only few columns are selected.

<details>
  <summary>Benchmarks</summary>

```
# Results (server side duration in ms)
V1: 5.5.0-2ffeaaa7e3a3cb1a386b7c49685842f42f810d70
V2: 5.5.0-f42107b17b09d7961c03a2a39f07f9fd32e56a2d

Q: select duration from uservisits where "sourceIP" = '25.193.131.52'
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |        0.440 ±    0.554 |      0.163 |      0.311 |      0.432 |      7.457 |
|   V2    |        0.416 ±    0.471 |      0.161 |      0.303 |      0.409 |      9.111 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               -   5.52%                           -   2.31%   
There is a 69.62% probability that the observed difference is not random, and the best estimate of that difference is 5.52%
The test has no statistical significance


System/JVM Metrics (durations in ms, byte-values in MB)
    |    YOUNG GC            |       OLD GC           |      HEAP         |     ALLOC     
    |  cnt      avg      max |  cnt      avg      max |  initial     used |     rate      total
 V1 |    0     0.00     0.00 |    0     0.00     0.00 |      268        0 |  4130.85      18419
 V2 |    0     0.00     0.00 |    0     0.00     0.00 |      268        0 |  3878.19      18359
    
Top allocation frames
  V1
    StreamSpliterators$WrappingSpliterator.initPartialTraversalState() total=16444653392, count=11
    ArrayUtil.growExact(...) total=1619107337, count=34
    ArrayUtil.growNoCopy(...) total=56496920, count=24
    DirectMethodHandle.allocateInstance(Object) total=24599032, count=12
    BufferRecycler.balloc(int) total=20689368, count=6
    ArrayList.grow(int) total=15330192, count=14
    CommonTokenFactory.create(...) total=13160168, count=4
    ReduceOps.makeRef(Collector) total=11284256, count=3
    HashMap.newNode(...) total=9150928, count=7
    HashMap.resize() total=8834248, count=6
  V2
    StreamSpliterators$WrappingSpliterator.initPartialTraversalState() total=16369635264, count=11
    ArrayUtil.growNoCopy(...) total=1603539336, count=28
    BufferRecycler.balloc(int) total=53136128, count=18
    DirectMethodHandle.allocateInstance(Object) total=31617608, count=14
    ArrayUtil.growExact(...) total=24585744, count=14
    ArrayList.grow(int) total=15946992, count=11
    ArrayUtil.growNoCopy(int[], int) total=14318232, count=4
    ArrayList.iterator() total=12753368, count=6
    AbstractByteBuf.duplicate() total=12748296, count=2
    ReduceOps$3.makeSink() total=12377448, count=4

Top frames (by count)
  V1
    ArrayUtil.growExact(...) total=1619107337, count=34
    ArrayUtil.growNoCopy(...) total=56496920, count=24
    ArrayList.grow(int) total=15330192, count=14
    DirectMethodHandle.allocateInstance(Object) total=24599032, count=12
    StreamSpliterators$WrappingSpliterator.initPartialTraversalState() total=16444653392, count=11
    ObjectName.construct(String) total=25264, count=11
    StreamSupport.stream(...) total=4479264, count=11
    SegmentTermsEnumFrame.<init>(...) total=11168, count=8
    HashMap.newNode(...) total=9150928, count=7
    ObjectName.setCanonicalName(...) total=14480, count=7
  V2
    ArrayUtil.growNoCopy(...) total=1603539336, count=28
    BufferRecycler.balloc(int) total=53136128, count=18
    ArrayUtil.growExact(...) total=24585744, count=14
    DirectMethodHandle.allocateInstance(Object) total=31617608, count=14
    StreamSpliterators$WrappingSpliterator.initPartialTraversalState() total=16369635264, count=11
    ArrayList.grow(int) total=15946992, count=11
    ObjectName.construct(String) total=23352, count=10
    SegmentTermsEnumFrame.<init>(...) total=11104, count=8
    StreamSupport.stream(...) total=6388800, count=8
    ArrayList.iterator() total=12753368, count=6

```

### CODE REVIEW

File: server/src/main/java/io/crate/execution/jobs/PKLookupTask.java [hunk 1]
```diff
@@ -98,6 +105,19 @@ public final class PKLookupTask extends AbstractTask {
...
+        sourceParser = new SourceParser();
+        for (Symbol symbol : toCollect) {
+            RefVisitor.visitRefs(
+                    symbol,
+                    ref -> {
+                        if (ref.column().isSystemColumn() == false && ref.granularity() == RowGranularity.DOC) {
+                            sourceParser.register(toSourceLookup(ref).column(), ref.valueType());
+                        }
+                    }
+            );
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/execution/jobs/PKLookupTask.java:118]: There is an edge case of selecting _doc and regular column in a single query.

Could you try updating a test with such scenario to verify that _doc can see all columns?

Something like (https://github.com/crate/crate/pull/14665/commits/908bbcc666681a8a930c4b75ccb3a95c78e818dd#diff-dc89f0bea3e02a3afbff1c08f14546a984e60278688658124f864be8ad2bd484R1468-R1469)?

If it fails, I guess we need to add all table column into `toCollect`?

Comment (seut) [reply server/src/main/java/io/crate/execution/jobs/PKLookupTask.java]: Good catch! I've pushed a fixup to verify and fix it.

**Review by BaurzhanSakhariev — APPROVED** [review]
👍

**Comment by seut** [comment]
retest this please

unrelated failure on `io.crate.integrationtests.MetadataTrackerITest.test_subscription_to_multiple_publications_should_not_stop_on_a_single_publication_drop`:
```
org.elasticsearch.common.io.stream.NotSerializableExceptionWrapper: operation_on_inaccessible_relation_exception: The relation "doc.t2" doesn't allow INSERT operations, because it is included in a logical replication subscription.
```
